x-pack/plugins/security_solution/public/timelines/components/timeline/discover_tab_content/index.tsx

```diff
@@ -23,37 +24,35 @@ export const DiscoverTabContent = () => {
     services: { customDataService: discoverDataService, discover, discoverFilterManager },
   } = useKibana();
 
-  const { useDiscoverMainRoute } = discover;
-
   const { CustomStatefulTopNavKqlQueryBar } = useGetStatefulQueryBar();
 
-  const getDiscoverLayout = useDiscoverMainRoute({
-    services: {
-      filterManager: discoverFilterManager,
-      data: discoverDataService,
-    },
-  });
-
-  const DiscoverMainRoute = getDiscoverLayout(history);
-
-  const customizationCallback: CustomizationCallback = useCallback(
-    ({ customizations, stateContainer }) => {
+  const customize: CustomizationCallback = useCallback(
+    ({ customizations }) => {
       customizations.set({
         id: 'search_bar',
-        CustomQueryBar: CustomStatefulTopNavKqlQueryBar,
-      });
-
-      customizations.set({
-        id: 'top_nav',
-        showBreadcrumbs: false,
+        CustomSearchBar: CustomStatefulTopNavKqlQueryBar,
       });
     },
     [CustomStatefulTopNavKqlQueryBar]
   );
 
+  const services = useMemo(
+    () => ({
+      filterManager: discoverFilterManager,
+      data: discoverDataService,
+    }),
+    [discoverDataService, discoverFilterManager]
+  );
+
+  const DiscoverContainer = discover.DiscoverContainer;
+
   return (
     <EmbeddedDiscoverContainer>
-      <DiscoverMainRoute isDev={false} customizationCallbacks={[customizationCallback]} />
+      <DiscoverContainer
+        services={services}
+        scopedHistory={history as ScopedHistory}
```

Comment:
Within Discover `setScopedHistory` expects a `ScopedHistory` type. The `useDiscoverMainRoute` hook was previously accepting a `history: any` param so it wasn't an issue, but now the prop is expecting the correct type so TS complains. This may be something to look into further since the types don't seem to match.

Reply:
I think typecasting it makes sense since we are getting history from `useHistory` hook. The initial value that provided to the route is `ScopedHistory` only. I will check if types can be corrected.